tests/clients/test_securitydata.py

```diff
@@ -1321,6 +1325,146 @@ def test_stream_file_by_md5_when_find_file_version_returns_204_status_code_raise
 
         assert e.value.args[0] == PDS_EXCEPTION_MESSAGE.format("mdhash")
 
+    def test_stream_file_by_md5_when_has_exact_match_calls_get_token_with_expected_params_and_streams_successfully(
```

Comment:
New tests start here, almost everything else in this PR is a result of refactoring `py42_response`